### PR TITLE
Feat/168 methodology integration

### PR DESCRIPTION
**Pull request by ArthurEncr**
# Description

Fetch MaleFemaleMethodology data from backend service then map options on the orchard Step

-  MaleFemaleMethodology data fetching
-  Male methodology data dynamically displayed
-  Female methodology data dynamically displayed
-  Special cases are accounted for: e.g. logepole pine having different options

Fixes # [168](https://github.com/bcgov/nr-spar/issues/168)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manually tested

### CODE REVIEW

**Comment by craigyu** [comment]
This will be reviewed after #261 is merged into `main`, #261 contains refactor of the `DropDownObj` and its util functions which will cause conflicts

**Comment by craigyu** [comment]
@ArthurEncr it's conflict resolution time 🥳

**Comment by ArthurEncr** [comment]
No more conflicts, it can now be revised :)

**Review by mgaseta — COMMENTED** [review]
LGTM! :rocket: :100:

**Review by mgaseta — APPROVED** [review]

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx [hunk 1]
```diff
@@ -352,31 +356,13 @@ const OrchardStep = ({
...
-                      labelText={orcharStepText.gameteSection.maleGametic.options.m5}
-                      value="M5"
-                    />
+                    {maleGameticOptions.map((item) => (
+                      <RadioButton
+                        id={`${item.code.toLowerCase()}-radio`}
+                        labelText={item.label}
+                        value={item.code}
+                      />
+                    ))}
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx:365]: Forgot about this, but there is a missing `key` prop within this map.

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx [hunk 2]
```diff
@@ -388,21 +374,13 @@ const OrchardStep = ({
...
-                      id="m3-radio"
-                      labelText={orcharStepText.gameteSection.maleGametic.options.m3}
-                      value="M3"
-                    />
+                    {maleGameticOptions.slice(0, -2).map((item) => (
+                      <RadioButton
+                        id={`${item.code.toLowerCase()}-radio`}
+                        labelText={item.label}
+                        value={item.code}
+                      />
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx:382]: Also here.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Forgot these two markups, sorry.

**Review by mgaseta — COMMENTED** [review]
LGTM! :100: :1st_place_medal:

**Review by mgaseta — APPROVED** [review]

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx [hunk 3]
```diff
@@ -388,21 +375,14 @@ const OrchardStep = ({
...
-                      id="m2-radio"
-                      labelText={orcharStepText.gameteSection.maleGametic.options.m2}
-                      value="M2"
-                    />
-                    <RadioButton
-                      id="m3-radio"
-                      labelText={orcharStepText.gameteSection.maleGametic.options.m3}
-                      value="M3"
-                    />
+                    {maleGameticOptions.slice(0, -2).map((item) => (
```
Comment (craigyu) [inline frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx:378]: How can we know for sure that the sliced objects are the correct ones>

Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx:378]: there will be a backend task to handle this

**Review by craigyu — APPROVED** [review]
femaleGametic is stored as an object(multiOptionObj) while maleGametic is stored as a string(code), it would be nice if they are both storing the code for consistency but this is ok for now

**Pull request merged**